src/fetchers/tarball-fetcher.js

```diff
@@ -254,14 +258,25 @@ export default class TarballFetcher extends BaseFetcher {
             ...headers,
           },
           buffer: true,
-          process: (req, resolve, reject) => {
+          process: (req, res, resolve, reject, queueForRetry) => {
```

Comment:
I'd keep the new parameter before the `resolve, reject` parameters to keep a more promise-like interface in the future.

```suggestion
          process: (req, res, queueForRetry, resolve, reject) => {
```

Reply:
This one is a little tricky, since it doesn't fully match Promise semantics. If it is retried, _neither_ resolve nor reject is called - it is only called after the next call to `process`. I'll see if there's a cleaner way to do this.